rabbitmq/consumer.go

```diff
@@ -80,86 +114,96 @@ func (c *RabbitMQConsumer) Run(ctx context.Context) error {
 		nil,
 	)
 	if err != nil {
-		return stacktrace.Propagate(err, "couldn't start consuming from channel")
+		return stacktrace.Propagate(err, "couldn't start consuming from RMQ channel")
 	}
 
 	err = c.handleDeliveries(ctx, deliveries)
-
-	return stacktrace.Propagate(err, "stopped consumer")
+	return stacktrace.Propagate(err, "failed/stopped handling RMQ consumer deliveries")
 }
 
 // nolint:gocognit
-func (c *RabbitMQConsumer) handleDeliveries(ctx context.Context, deliveries <-chan amqp.Delivery) error {
-	for d := range deliveries {
-		// nolint:godox
-		// TODO: until we add better logging level conditions
-		//c.logger.Debug(
-		//	"msg delivered",
-		//	zap.Uint64("tag", d.DeliveryTag),
-		//	zap.ByteString("body", d.Body),
-		//)
-
-		c.metric.ObserveMsgDelivered()
-
-		acknowledgement, err := c.handler.ReceiveMessage(ctx, d.Body)
-		if err != nil {
-			return stacktrace.Propagate(err, "handler returned error")
+func (c *Consumer) handleDeliveries(
+	ctx context.Context,
+	deliveries <-chan amqp.Delivery,
+) error {
+	for {
+		select {
+		case <-ctx.Done():
+			c.logger.Warn("RMQ handler stopping")
+			return nil
```

Comment:
Shouldn't we return `ctx.Err()` here?

Reply:
No, the `ctx.Err` would be pointless, since it'll be a `cancellation error`